### PR TITLE
Provide structured suggestion for `hashmap[idx] = val`

### PR DESCRIPTION
**Pull request by estebank**

### CODE REVIEW

File: compiler/rustc_borrowck/src/diagnostics/mutability_errors.rs [hunk 1]
```diff
@@ -632,13 +635,115 @@ impl<'a, 'tcx> MirBorrowckCtxt<'a, 'tcx> {
...
+                                        val.span.shrink_to_hi().with_hi(index.span.lo()),
+                                        ".insert(".to_string(),
+                                    ),
+                                    (
+                                        index.span.shrink_to_hi().with_hi(rv.span.lo()),
+                                        ", ".to_string(),
+                                    ),
+                                    (rv.span.shrink_to_hi(), ")".to_string()),
+                                ],
+                                vec![ // val.get_mut(index) = rv;
```
Comment (fmease) [inline compiler/rustc_borrowck/src/diagnostics/mutability_errors.rs:684]: `get_mut` returns an `Option` while `rv` may be of any type `V`. Further, the LHS, even if `unwrap()`'ed is not a place expression and it first needs to be dereferenced. This suggestion does not compile in any case.

Quoting snippets from the UI tests:

```patch
- map.get_mut("peter") = "0".to_string();
+ *map.get_mut("peter").unwrap() = "0".to_string();
```

Still, this is very likely not what the user intended to write but at least it compiles.

**Review by davidtwco — APPROVED** [review]
r=me after resolving @fmease and @compiler-errors' comments

**Review by davidtwco — APPROVED** [review]

**Pull request merged**